src/main/java/org/elasticsearch/index/query/BaseQueryBuilder.java

```diff
@@ -68,5 +69,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    public abstract Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException;
```

Comment:
1) sounds good, that's where I was going too. let's try and not duplicate parser names that we already have.